File: lark/src/main/java/cc/omoz/lark/ctrls/IndexCtrl.java
package cc.omoz.lark.ctrls;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.ModelAndView;

/**
 * index控制器
 *
 * @author ThinkTik
 */
@RestController
@RequestMapping("/")
public class IndexCtrl {
    /**
     * 应用程序名
     */
    @Value("${spring.application.name}")
    private String appName;
    /**
     * 描述
     */
    @Value("${description}")
    private String description;
    /**
     * index 主页URL
     */
    @Value("${indexUrl}")
    private String indexUrl;

    /**
     * redirect到index
     *
     * @return index页面
     */
    @GetMapping(value = {"", "/index"})
    public ModelAndView index() {
        ModelAndView mav = new ModelAndView("index");
        mav.addObject("appName", appName);
        mav.addObject("description", description);
        mav.addObject("indexUrl", indexUrl);
        return mav;
    }
}

File: eagle/src/main/scala/cc/omoz/eagle/ctrls/IndexCtrl.scala
package cc.omoz.eagle.ctrls

import cc.omoz.facade.omoz.dto.impl.OmozHttpResponse
import org.springframework.beans.factory.annotation.Value
import org.springframework.web.bind.annotation.{GetMapping, RestController}

import java.util

/**
 * index控制器
 *
 * @author ThinkTik
 */
@RestController
class IndexCtrl(
                 @Value("${spring.application.name}") val appName: String,
                 @Value("${description}") val description: String,
                 @Value("${indexUrl}") val indexUrl: String
               ) {
  @GetMapping(value = Array("/index"))
  def index(): OmozHttpResponse[util.Map[String, String]] = {
    val response = new OmozHttpResponse[util.Map[String, String]]()
    val data = new util.HashMap[String, String]()
    data.put("appName", appName)
    data.put("description", description)
    response.setCode(200)
    response.setMessage("success")
    response.setData(data)

    response
  }
}